collectors/osidb_scheduler/collectors.py

```diff
@@ -0,0 +1,67 @@
+"""
+OSIDB Collectors
+"""
+from celery.utils.log import get_task_logger
+from django.contrib.contenttypes.models import ContentType
+from django.db.models import OuterRef, Q, Subquery
+from django.db.models.functions import Cast
+
+from collectors.framework.models import Collector
+from osidb.mixins import Alert
+
+logger = get_task_logger(__name__)
+
+
+class StaleAlertCollector(Collector):
+    """
+    Stale Alert Collector
+    """
+
+    def collect(self):
+        """
+        collector run handler
+
+        On every run, this collector will check if the alert is
+        still valid by comparing the creation time of the alert
+        with the validation time of the Model.
+
+        If the creation time of the alert is older than
+        the validation time of the Model,
+        the alert is considered stale and will be deleted.
+        """
+        content_types = ContentType.objects.filter(
+            id__in=Alert.objects.values_list("content_type", flat=True).distinct()
+        )
+
+        logger.info(
+            f"Searching for stale alerts in {content_types.count()} content types"
+        )
+
+        query = Q()
+
+        for content_type in content_types:
+            model_class = content_type.model_class()
+
+            subquery = Subquery(
+                model_class.objects.filter(
+                    pk=Cast(OuterRef("object_id"), output_field=model_class._meta.pk)
+                ).values("last_validated_dt")[:1]
```

Comment:
@MrMarble is this subquery sorted by default?

Reply:
The subqueries are sorted by the default `Model.Meta.ordering` fields, but I think you're right, it should be sorted by the `last_validated_dt` field